pkg/connector/matrix.go

```diff
@@ -75,7 +76,23 @@ func (t *TelegramClient) transferMediaToTelegram(ctx context.Context, content *e
 	filename := getMediaFilename(content)
 	err := t.main.Bridge.Bot.DownloadMediaToFile(ctx, content.URL, content.File, false, func(f *os.File) (err error) {
 		uploadFilename := f.Name()
-		if sticker && content.Info != nil && (content.Info.MimeType != "video/webm" && content.Info.MimeType != "application/x-tgsticker") {
+		if sticker && content.Info != nil && (content.Info.MimeType == "image/png" || content.Info.MimeType == "image/jpeg") {
+			tempFile, err := os.CreateTemp("", "telegram-sticker-*")
```

Comment:
Image conversion generally shouldn't require a temp file 🤔

Reply:
It's already decoding the file into memory, just write it back into the same file

Reply:
Considered that but thought it was evil :sweat_smile:

Reply:
Can't truncate the temp file that is being passed to the callback, go probably doesn't open it with os.O_TRUNC.

Reply:
can't you truncate without reopening 🤔

the file passed to download is already a temp file

Reply:
oh nevermind, it's not necessarily a temp file unless the writable flag is passed

Reply:
For posterity the behavior of the file in the callback depends on implementation details and the safest bet is to just create our own, using writable flag would work but that may cause an unnecessary copy of the file being created

There was a bug in our other implementation that made the writable flag not work as expected which threw me off when I tried to refactor this initially.